src/render.rs

```diff
@@ -849,6 +857,52 @@ impl Renderer {
             };
         }
 
+        // draw background (if any)
+        match scene.background {
+            Background::Texture(ref texture) => {
+                // TODO: Reduce code duplication (see drawing debug quads)
+                self.encoder.update_constant_buffer(
+                    &self.quad_buf,
+                    &QuadParams {
+                        rect: [-1.0, -1.0, 1.0, 1.0],
+                        depth: 1.0,
+                    },
+                );
+                let slice = gfx::Slice {
+                    start: 0,
+                    end: 4,
+                    base_vertex: 0,
+                    instances: None,
+                    buffer: gfx::IndexBuffer::Auto,
+                };
+                let data = quad_pipe::Data {
+                    params: self.quad_buf.clone(),
+                    resource: texture.to_param().0.raw().clone(),
+                    sampler: texture.to_param().1,
+                    target: self.out_color.clone(),
+                    depth_target: self.out_depth.clone(),
+                };
+                self.encoder.draw(&slice, &self.pso_quad, &data);
+            }
+            Background::Skybox(ref cubemap) => {
+                let slice = gfx::Slice {
```

Comment:
looks like we can move this slice declaration up and share

Reply:
Sounds reasonable